maps-compose/src/main/java/com/google/maps/android/compose/streetview/StreetView.kt

```diff
@@ -0,0 +1,145 @@
+package com.google.maps.android.compose.streetview
+
+import android.content.ComponentCallbacks
+import android.content.res.Configuration
+import android.os.Bundle
+import android.util.Log
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Composition
+import androidx.compose.runtime.CompositionContext
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCompositionContext
+import androidx.compose.runtime.rememberUpdatedState
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalLifecycleOwner
+import androidx.compose.ui.viewinterop.AndroidView
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleEventObserver
+import com.google.android.gms.maps.StreetViewPanoramaOptions
+import com.google.android.gms.maps.StreetViewPanoramaView
+import com.google.android.gms.maps.model.StreetViewPanoramaCamera
+import com.google.android.gms.maps.model.StreetViewPanoramaOrientation
+import com.google.maps.android.compose.disposingComposition
+import com.google.maps.android.ktx.MapsExperimentalFeature
+import com.google.maps.android.ktx.awaitStreetViewPanorama
+
+@MapsExperimentalFeature
+@Composable
+public fun StreetView(
+    modifier: Modifier = Modifier,
+    cameraPositionState: StreetViewCameraPositionState = rememberStreetViewCameraPositionState(),
+    streetViewPanoramaOptionsFactory: () -> StreetViewPanoramaOptions = {
+        StreetViewPanoramaOptions()
+    },
+    isPanningGesturesEnabled: Boolean = true,
+    isStreetNamesEnabled: Boolean = true,
+    isUserNavigationEnabled: Boolean = true,
+    isZoomGesturesEnabled: Boolean = true,
+    onClick: (StreetViewPanoramaOrientation) -> Unit = {},
+    onLongClick: (StreetViewPanoramaOrientation) -> Unit = {},
+) {
+    val context = LocalContext.current
+    val streetView =
+        remember { StreetViewPanoramaView(context, streetViewPanoramaOptionsFactory()) }
```

Comment:
This could leak a Context, since only the `Context` in the first composition will be used. Same problem exists elsewhere in the library.

Reply:
Thanks - yeah this should key against `Context`

Reply:
Why not use the Context given by AndroidView's `factory`?

Reply:
Thanks. Yeah that's the correct one to use

Reply:
Actually... We can't use that as the `StreetViewPanoramaView` has to be constructed outside of the `AndroidView`'s factory lambda since the `StreetViewPanorama`. It's a bit of a limitation currently with `AndroidView` and breaks guidance on not `remember`ing a View. However, there's no way around this given that it has to be hoisted up.